src/components/browse/components/table-commons/HeadersRow.js

```diff
@@ -199,16 +214,31 @@ export class HeadersRow extends React.PureComponent {
      * before calling `props.sortByFxn`.
      */
     sortByField(field){
-        const { sortColumns, sortBy } = this.props;
+        const { sortColumns, sortBy, columnDefinitions } = this.props;
         const [{ column = null, direction = "desc" } = {}] = sortColumns || [];
 
         const trimmedColumn = HeadersRow.getTrimmedColumn(column);
 
+        let initialSort;
+        if (columnDefinitions) {
+            const [colDef] = _.filter(columnDefinitions, function (item) { return item.field == field; });
```

Comment:
Use `find` or similar instead, it'll stop searching as soon as finds proper result instead of iterating over entire list (even if found matching result).

```suggestion
            const colDef = columnDefinitions.find(function (item) { return item.field == field; });
```

Reply:
Also instead of memoizing `getSortDirectionBySchemaFieldType`, would maybe create memoized func along lines of the following (to memoize the `.find`):
```
getInitialSort(columnDefinitions){
    const colDef = columnDefinitions.find(function (item) { return item.field == field; });
    if (colDef && (colDef.type || colDef.initial_sort)) {
        return colDef.initial_sort || getSortDirectionBySchemaFieldType(colDef.type);
    }
    return null;
}
```